### PR TITLE
Fix MMRest crash when doing ctrl+Z

### PR DESCRIPTION
**Pull request by mike-spa**
Resolves: #14350

Just a missing null-check, it seems.

### CODE REVIEW

**Review by oktophonie — APPROVED** [review]

File: src/notation/internal/notationselectionrange.cpp [hunk 1]
```diff
@@ -221,7 +221,7 @@ const
         mu::engraving::System* currentSegmentSystem = segment->measure()->system();
 
         mu::engraving::Segment* nextSegment = segment->next1MMenabled();
-        mu::engraving::System* nextSegmentSystem = nextSegment->measure()->system();
+        mu::engraving::System* nextSegmentSystem = nextSegment ? nextSegment->measure()->system() : segment->measure()->system();
```
Comment (RomanPudashkin) [inline src/notation/internal/notationselectionrange.cpp:224]: Why is it null?

Comment (mike-spa) [reply src/notation/internal/notationselectionrange.cpp:224]: I just noticed that Segment::next1MM() (which is called by Segment::next1MMenabled()) can explicitely return a null, so its output should be null-checked. Why exactly it returns null in this case, I'm not sure, I can look into it

Comment (RomanPudashkin) [reply src/notation/internal/notationselectionrange.cpp:224]: >Why exactly it returns null in this case, I'm not sure

We should find out why it returns null

Comment (mike-spa) [reply src/notation/internal/notationselectionrange.cpp]: It seems that we were skipping the exit condition of the loop, so it was running until the end of the score (where a null return is normal). Now it should be good

Comment (RomanPudashkin) [reply src/notation/internal/notationselectionrange.cpp]: 1. You should create a temp variable for this expression outside the loop
`rangeEndSegment->tick()`

2. How is this change related to Ctrl+Z? I don't get it from the commit message

Also, please notice that there is some mess in the commit message

Screen Shot 2022-11-22 at 11 35 46

Comment (mike-spa) [reply src/notation/internal/notationselectionrange.cpp]: I think I was finally able to find the root problem: when doing ctrl+Z we were tying to reset the selection, but we were computing the wrong end-segment in `Selection::updateSelectedElements()`, which was messing up the rest. Now it's fixed (but I still changed the exit condition of the loop to make it safer).

**Review by RomanPudashkin — APPROVED** [review]

**Review by oktophonie — APPROVED** [review]

**Comment by RomanPudashkin** [comment]
@mike-spa please open the same PR, but for the 4.0_rc branch

**Pull request merged**